### PR TITLE
Feature/publication

### PR DESCRIPTION
**Pull request by Savageman**

### CODE REVIEW

**Comment by sdrdis** [comment]
If I am not wrong, we must name function in camelCase ?

File: framework/classes/orm/behaviour/publishable.php [hunk 1]
```diff
@@ -61,20 +124,46 @@ public function before_query(&$options)
...
+            $published_key = null;
+            $published_value = null;
+
             foreach ($where as $k => $w) {
                 if (is_int($k)) {
-                    $keys = array_keys($w);
-                    if (count($w) == 1 && $keys[0] == 'published') {
-                        $where[$k] = array($this->_properties['publication_bool_property'] => $w[$keys[0]]);
+                    reset($w);
+                    // array('published' => 1);
```
Comment (sdrdis) [inline framework/classes/orm/behaviour/publishable.php]: some debugging comment I think :)

File: framework/classes/orm/behaviour/publishable.php [hunk 2]
```diff
@@ -61,20 +124,46 @@ public function before_query(&$options)
...
+                    if (count($w) == 1 && key($w) == 'published') {
+                        $published_key = $k;
+                        $published_value = (bool) current($w);
                     }
 
-                    if (count($w) > 1 && $w[0] == 'published') {
-                        $w[0] = $this->_properties['publication_bool_property'];
-                        $where[$k] = $w;
+                    // array('published', $value);
```
Comment (sdrdis) [inline framework/classes/orm/behaviour/publishable.php]: Same here

File: framework/classes/orm/behaviour/publishable.php [hunk 3]
```diff
@@ -12,48 +12,111 @@
...
-    public static function publication_status($item)
+    public function publication_status($item)
     {
-        $published = $item->published();
-        if ($published === true) {
+
+        $property = $this->_properties['publication_state_property'];
+        $status = $item->get($property);
+        if ($status == 0) {
+            return '<img class="publication_status" src="static/novius-os/admin/novius-os/img/icons/status-red.png"> '.__('Not published');
```
Comment (sdrdis) [inline framework/classes/orm/behaviour/publishable.php]: Could we move all those string into views in order to improve etendability ?

File: framework/classes/orm/behaviour/publishable.php (hunk 1 above)
Comment (Savageman) [inline framework/classes/orm/behaviour/publishable.php]: Actually no. It's a bit hard to see only with the diff, and my comment could be clearer, but it means that this portion of code handle the case when the where is written using key => value.

File: framework/classes/orm/behaviour/publishable.php (hunk 3 above)
Comment (felixgilles) [inline framework/classes/orm/behaviour/publishable.php]: I agree

File: framework/classes/orm/behaviour/publishable.php [hunk 4]
```diff
@@ -12,48 +12,111 @@
...
+        if (!empty($end)) {
+            return '<img class="publication_status" src="static/novius-os/admin/novius-os/img/icons/status-green.png"> '.strtr(__('Published until {{date}}'), array(
+                '{{date}}' => \Date::forge(strtotime($end))->format(__('date_format.normal')),
+            ));
+        }
+
+        return '<img class="publication_status" src="static/novius-os/admin/novius-os/img/icons/status-green.png"> '.__('Published');
+    }
+
+    public function planification_status($item)
```
Comment (felixgilles) [inline framework/classes/orm/behaviour/publishable.php]: All new method must be in camelCase

**Comment by Savageman** [comment]
Here you are :)

**Pull request merged**